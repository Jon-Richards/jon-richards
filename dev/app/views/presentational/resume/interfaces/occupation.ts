import { Responsibility } from './responsibility';

export interface Occupation {
  id: string;
  jobTitle: string;
  startDate: string;
  endDate: string;
  organization: string;
  responsibilities: Responsibility[];
}
