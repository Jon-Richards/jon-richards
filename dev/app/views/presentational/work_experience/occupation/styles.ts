import { css } from '@emotion/core';
import { Fonts } from 'Config/styles/fonts';
import { Dimensions } from 'Config/styles/dimensions';
import { Colors } from 'Config/styles/colors';

export const STYLES = {
  jobTitle: css`
    font-family: ${Fonts.OPEN_SANS.fontFamily};
    font-size: 1.4rem;
    font-weight: ${Fonts.OPEN_SANS.fontStyles.regular.weights.light};
    margin-bottom: calc(${Dimensions.whitespace} / 2);
  `,
  organization: css`
    font-family: ${Fonts.OPEN_SANS.fontFamily};
    font-size: 1rem;
    font-weight: ${Fonts.OPEN_SANS.fontStyles.regular.weights.regular};
    margin-bottom: 1rem;
  `,
  date: css`
    display: inline-block;
    font-family: ${Fonts.OPEN_SANS.fontFamily};
    font-style: ${Fonts.OPEN_SANS.fontStyles.italic.name};
    font-weight: ${Fonts.OPEN_SANS.fontStyles.italic.weights.regular};
    color: ${Colors.SWATCH_1.withLightness(70).value};
    padding-bottom: calc(${Dimensions.whitespace} / 2);
    margin-bottom: calc(${Dimensions.whitespace} / 2);
    border-bottom: thin dotted ${Colors.SWATCH_2.withLightness(30).value};
  `,
  description: css`
    padding-bottom: calc(${Dimensions.whitespace} * 0.5);
    margin-bottom: ${Dimensions.whitespace};
    border-bottom: thin solid ${Colors.SWATCH_2.withLightness(30).value};
    &:last-of-type {
      border-bottom: none;
      padding-bottom: 0;
      margin-bottom: 0;
    }
  `,
  responsibility: css`
    display: list-item;
    list-style-type: circle;
    font-family: ${Fonts.OPEN_SANS.fontFamily};
    font-weight: ${Fonts.OPEN_SANS.fontStyles.regular.weights.regular};
    line-height: 1.6rem;
    padding-left: 0.5rem;
    margin-left: 2rem;
    margin-bottom: calc(${Dimensions.whitespace} / 2);

    &::before {
      right: 1rem;
      opacity: 0.5;
    }
  `
};
