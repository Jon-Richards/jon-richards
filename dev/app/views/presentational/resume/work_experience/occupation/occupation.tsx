/** @jsx jsx */

import * as React from 'react';
import { jsx } from '@emotion/core';
import { STYLES } from './styles';
import { Occupation as IOccupation } from '../../interfaces/occupation';
import { Responsibility } from '../../interfaces/responsibility';
import { UnorderedList } from 'Views/presentational/lists';

export function Occupation(props: OccupationProps): JSX.Element {
  return (
    <React.Fragment>
      <dt
        data-testid="occupation__job-title"
        css={STYLES.jobTitle}
      >
        {props.jobTitle}
      </dt>
      <dd
        data-testid="occupation__job-description"
        css={STYLES.description}
      >
        <p css={STYLES.organization}>{props.organization}</p>
        <p css={STYLES.date}>{props.startDate} - {props.endDate}</p>
        <UnorderedList
          title="Responsibilities"
          items={mapResponsabilities(props.responsibilities)}
        />
      </dd>
    </React.Fragment>
  );
}

export type OccupationProps = Omit<IOccupation, 'id'>;

function mapResponsabilities(respnsabilities: Responsibility[]) {
  const result = respnsabilities.map(responsibility => (
    {
      id: responsibility.id,
      content: responsibility.description
    }
  ));
  return result;
}
