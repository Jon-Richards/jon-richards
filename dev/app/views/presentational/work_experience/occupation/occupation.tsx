/** @jsx jsx */

import * as React from 'react';
import { jsx } from '@emotion/core';
// import { List } from 'Views/presentational/list';
import { STYLES } from './styles';

export function Occupation(props: OccupationProps): JSX.Element {
  return (
    <React.Fragment>
      <dt
        data-testid="occupation__job-title"
        css={STYLES.jobTitle}
      >
        {props.jobTitle}
      </dt>
      <dd
        data-testid="occupation__job-description"
        css={STYLES.description}
      >
        <p css={STYLES.organization}>{props.organization}</p>
        <p css={STYLES.date}>{props.startDate} - {props.endDate}</p>
        <ul>
          {mapResponsabilities(props.responsabilities)}
        </ul>
      </dd>
    </React.Fragment>
  );
}

export type OccupationProps = {
  organization: string,
  startDate: string,
  endDate: string,
  jobTitle: string,
  responsabilities: responsibility[];
}

type responsibility = {
  id: string;
  text: string
};

function mapResponsabilities(respnsabilities: responsibility[]): JSX.Element {
  const result = respnsabilities.map(responsibility =>
    <li
      key={responsibility.id}
      data-testid="occupation__responsibility"
      css={STYLES.responsibility}
    >
      {responsibility.text}
    </li>
  );
  return <React.Fragment>{result}</React.Fragment>
}
