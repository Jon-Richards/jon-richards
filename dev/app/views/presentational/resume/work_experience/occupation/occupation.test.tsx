import * as React from 'react';
import { render } from 'enzyme';
import { Occupation, OccupationProps } from './occupation';

describe('The Occupation component.', () => {
  it('Matches the snapshot.', () => {
    const responsibilities: OccupationProps['responsibilities'] = [
      { id: 'one', description: 'Foo' },
      { id: 'two', description: 'Bar' },
    ]

    const wrapper = render(
      <dl>
        <Occupation
          jobTitle="Sr. UI Developer"
          organization="Some Organization"
          startDate="Jan 1, 2021"
          endDate="Dec 31, 2021"
          responsibilities={responsibilities}
        />
      </dl>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Renders a responsibility for each on provided', () => {
    const responsibilities: OccupationProps['responsibilities'] = [
      {
        id: 'one',
        description: <p data-testid="occupation__responsibility">Foo</p>
      },
      {
        id: 'two',
        description: <p data-testid="occupation__responsibility">Bar</p>
      },
    ]

    const wrapper = render(
      <dl>
        <Occupation
          jobTitle="Sr. UI Developer"
          organization="Some Organization"
          startDate="Jan 1, 2021"
          endDate="Dec 31, 2021"
          responsibilities={responsibilities}
        />
      </dl>
    );

    const testHook = '[data-testid="occupation__responsibility"]';

    expect(wrapper.find(testHook).length).toBe(2);
  });
});
