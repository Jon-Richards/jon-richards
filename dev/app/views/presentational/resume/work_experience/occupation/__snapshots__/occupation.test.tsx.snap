// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Occupation component. Matches the snapshot. 1`] = `
.emotion-0 {
  font-family: "Open Sans",sans-serif;
  font-size: 1.4rem;
  font-weight: 300;
  margin-bottom: calc(2rem / 2);
}

.emotion-6 {
  padding-bottom: calc(2rem * 0.5);
  margin-bottom: 2rem;
  border-bottom: thin solid hsla(48,7%,30%,1);
}

.emotion-6:last-of-type {
  border-bottom: none;
  padding-bottom: 0;
  margin-bottom: 0;
}

.emotion-1 {
  font-family: "Open Sans",sans-serif;
  font-size: 1rem;
  font-weight: 400;
  margin-bottom: 1rem;
}

.emotion-2 {
  display: inline-block;
  font-family: "Open Sans",sans-serif;
  font-style: italic;
  font-weight: 400;
  color: hsla(0,0%,70%,1);
  padding-bottom: calc(2rem / 2);
  margin-bottom: calc(2rem / 2);
  border-bottom: thin dotted hsla(48,7%,30%,1);
}

.emotion-5 {
  list-style-type: circle;
}

.emotion-3 {
  display: list-item;
  font-family: "Open Sans",sans-serif;
  font-weight: 400;
  line-height: 1.6rem;
  padding-left: 0.5rem;
  margin-left: 2rem;
  margin-bottom: calc(2rem / 2);
}

.emotion-3::before {
  right: 1rem;
  opacity: 0.5;
}

<dl>
  <dt
    class="emotion-0"
    data-testid="occupation__job-title"
  >
    Sr. UI Developer
  </dt>
  <dd
    class="emotion-6"
    data-testid="occupation__job-description"
  >
    <p
      class="emotion-1"
    >
      Some Organization
    </p>
    <p
      class="emotion-2"
    >
      Jan 1, 2021 - Dec 31, 2021
    </p>
    <ul
      class="emotion-5"
      data-testid="unordered-list"
      title="Responsibilities"
    >
      <li
        class="emotion-3"
        data-testid="unordered-list__item"
      >
        Foo
      </li>
      <li
        class="emotion-3"
        data-testid="unordered-list__item"
      >
        Bar
      </li>
    </ul>
  </dd>
</dl>
`;
